packages/core/src/browser/common-frontend-contribution.ts

```diff
@@ -281,6 +291,8 @@ export const supportPaste = browser.isNative || (!browser.isChrome && document.q
 
 export const RECENT_COMMANDS_STORAGE_KEY = 'commands';
 
+const PINNED_CLASS = 'theia-mod-pinned';
```

Comment:
@sh7dm something to try, rather than include the `svg` for each icon, we can simply use `codicons` directly:

```
codicon codicon-pinned
```

We have other examples in the codebase:

https://github.com/eclipse-theia/theia/search?q=codicon

Reply:
What about integration with dirty classes and positioning?

Reply:
Same idea as the following (using the [unicode](https://github.com/microsoft/vscode-codicons/blob/main/dist/codicon.csv)):


```css
.theia-search-button.codicon-filter:not(.filter-active):before {
    content: "\eb85";
}

.theia-search-button.codicon-filter.filter-active:before {
    content: "\eb83";
}
```

It is something to at least try and see if it works :)

Reply:
Weird, those codicons don't work, maybe codicon package is out-of-date?
![image](https://user-images.githubusercontent.com/19504461/122787509-143b9380-d2be-11eb-855a-c324c084febe.png)

Reply:
> Weird, those codicons don't work, maybe codicon package is out-of-date?

It looks like `pin` is supported, but `pin-dirty` would work once we update monaco (https://github.com/eclipse-theia/theia/pull/9154) which contributes the codicons.

Reply:
So, what can we do? I can keep SVGs and add a TODO to them about that update.f

Reply:
> So, what can we do? I can keep SVGs and add a TODO to them about that update.

I think that would be fine, the monaco merge should happen fairly soon but doing a refactor from svg to codicons can be done at a later time 👍